symfony/plugins/orangehrmAdminPlugin/Api/WorkShiftAPI.php

```diff
@@ -0,0 +1,257 @@
+            new ParamRule(
+                CommonParams::PARAMETER_ID,
+                new Rule(Rules::POSITIVE)
+            ),
+        );
+    }
+
+    /**
+     * @inheritDoc
+     * @throws DaoException
+     */
+    public function create(): EndpointResult
+    {
+        $workShift = $this->saveWorkShift();
+        return new EndpointResourceResult(WorkShiftModel::class, $workShift);
+    }
+
+    /**
+     * @return WorkShift
+     * @throws RecordNotFoundException
+     * @throws DaoException
+     */
+    public function saveWorkShift(): WorkShift
+    {
+        $workShiftId = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_ID
+        );
+
+        $workShiftName = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_NAME);
+        $workShiftHoursPerDay = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_HOURS_PER_DAY
+        );
+        $workShiftStartTime = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_START_TIME
+        );
+        $workShiftEndTime = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_END_TIME
+        );
+        $empNumbers = $this->getRequestParams()->getArray(
+            RequestParams::PARAM_TYPE_BODY,
+            CommonParams::PARAMETER_EMP_NUMBERS
+        );
+
+        if (!empty($workShiftId)) {
+            $workShift = $this->getWorkShiftService()->getWorkShiftById($workShiftId);
+            if ($workShift == null) {
+                throw new RecordNotFoundException();
+            } else {
+                $workShift->setName($workShiftName);
+                $workShift->setHoursPerDay($workShiftHoursPerDay);
+                $workShift->setStartTime(new DateTime($workShiftStartTime));
+                $workShift->setEndTime(new DateTime($workShiftEndTime));
+                return $this->getWorkShiftService()
+                    ->getWorkShiftDao()
+                    ->updateWorkShift($workShift, $empNumbers);
+            }
+        } else {
+            $workShift = new WorkShift();
+        }
+        $workShift->setName($workShiftName);
+        $workShift->setHoursPerDay($workShiftHoursPerDay);
+        $workShift->setStartTime(new DateTime($workShiftStartTime));
+        $workShift->setEndTime(new DateTime($workShiftEndTime));
+
+        return $this->getWorkShiftService()
+            ->getWorkShiftDao()
+            ->saveWorkShift($workShift, $empNumbers);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                self::PARAMETER_NAME,
+                new Rule(Rules::STRING_TYPE),
+                new Rule(Rules::REQUIRED),
```

Comment:
no need this required rule. since by default param rules required, remove in below rules

Reply:
done